File: FFRv2/app/src/main/java/com/devingotaswitch/rankings/sources/ParseWalterFootball.java
package com.devingotaswitch.rankings.sources;

import com.devingotaswitch.rankings.domain.Rankings;
import com.devingotaswitch.rankings.domain.RosterSettings;
import com.devingotaswitch.rankings.domain.ScoringSettings;
import com.devingotaswitch.utils.Constants;
import com.devingotaswitch.utils.JsoupUtils;
import com.devingotaswitch.utils.ParsingUtils;

import java.io.IOException;
import java.util.List;

public class ParseWalterFootball {
    public static void wfRankings(Rankings rankings)
            throws IOException {
        RosterSettings r = rankings.getLeagueSettings().getRosterSettings();
        ScoringSettings s = rankings.getLeagueSettings().getScoringSettings();
        if (r.getQbCount() > 1 || (r.getFlex() != null && r.getFlex().getQbrbwrteCount() > 0)) {
            wfRankingsHelper(rankings,
                    "http://walterfootball.com/fantasycheatsheet/" + Constants.YEAR_KEY + "/twoqb");
        } else if (s.getReceptions() > 0) {
            wfRankingsHelper(rankings,
                    "http://walterfootball.com/fantasycheatsheet/" + Constants.YEAR_KEY + "/ppr");
        } else {
            wfRankingsHelper(rankings,
                    "http://walterfootball.com/fantasycheatsheet/" + Constants.YEAR_KEY + "/traditional");
        }
    }

    private static void wfRankingsHelper(Rankings rankings, String url)
            throws IOException, ArrayIndexOutOfBoundsException {
        List<String> perPlayer = JsoupUtils.parseURLWithUA(url,
                "ol.fantasy-board div li");
        String[][] all = new String[perPlayer.size()][];
        for (int i = 0; i < perPlayer.size(); i++) {
            all[i] = perPlayer.get(i).split(", ");
            String playerName = all[i][0];
            String pos = "";
            double val = 0.0;
            if (!perPlayer.get(i).contains("DEF")) {
                pos = all[i][1];
            } else {
                playerName += " D/ST";
                pos = Constants.DST;
            }
            val = Double
                    .parseDouble(perPlayer.get(i).split("\\$")[1].split(" ")[0]);
            String team = all[i][2].split("\\. ")[0];
            rankings.processNewPlayer(ParsingUtils.getPlayerFromRankings(playerName, team, pos, val));
        }
    }
}

File: FFRv2/app/src/main/java/com/devingotaswitch/rankings/sources/ParseWalterFootball.kt
package com.devingotaswitch.rankings.sources

import com.devingotaswitch.rankings.domain.Rankings
import com.devingotaswitch.utils.Constants
import com.devingotaswitch.utils.JsoupUtils.parseURLWithUA
import com.devingotaswitch.utils.ParsingUtils.getPlayerFromRankings
import java.io.IOException

object ParseWalterFootball {
    @Throws(IOException::class)
    fun wfRankings(rankings: Rankings) {
        val r = rankings.getLeagueSettings().rosterSettings
        val s = rankings.getLeagueSettings().scoringSettings
        if (r.qbCount > 1 || r.flex != null && r.flex!!.qbrbwrteCount > 0) {
            wfRankingsHelper(rankings,
                    "http://walterfootball.com/fantasycheatsheet/" + Constants.YEAR_KEY + "/twoqb")
        } else if (s.receptions > 0) {
            wfRankingsHelper(rankings,
                    "http://walterfootball.com/fantasycheatsheet/" + Constants.YEAR_KEY + "/ppr")
        } else {
            wfRankingsHelper(rankings,
                    "http://walterfootball.com/fantasycheatsheet/" + Constants.YEAR_KEY + "/traditional")
        }
    }

    @Throws(IOException::class, ArrayIndexOutOfBoundsException::class)
    private fun wfRankingsHelper(rankings: Rankings, url: String) {
        val perPlayer = parseURLWithUA(url,
                "ol.fantasy-board div li")
        var i = 0
        while (i < perPlayer.size) {
            // Rows are name, pos, team. bye $x
            val row: Array<String> = perPlayer[i].split(", ").toTypedArray()
            var playerName = row[0]
            var pos: String
            if (!perPlayer[i].contains("DEF")) {
                pos = row[1]
            } else {
                playerName += " D/ST"
                pos = Constants.DST
            }
            var aucVal = row[2].split("\\$".toRegex())[1].split(" ")[0].toDouble()
            if (aucVal < 0.0) {
                // For whatever reason, had players going as low as -$87
                aucVal = 1.0
            }
            val team = row[2].split("\\. ".toRegex())[0]
            rankings.processNewPlayer(getPlayerFromRankings(playerName, team, pos, aucVal))
            i ++
        }
    }
}